### PR TITLE
2305 Add Snapster class

### PR DESCRIPTION
**Pull request by BlimpDev**
To encapsulate functionality within a Snapster class so that we don't have to litter React components with business logic and users of the module have an easy to understand and use api.

- Unit tests for Snapster.
- Add new Snapster class.
- Create setup method that accepts boxes property with x, y, width, height, and properties.
- Create snap method that does the rendering and snapping and returns x and y values.
- Create clear method that clears the grid edges.

Renamed Snapper class to Snap to not confuse with the Snapster class.

### CODE REVIEW

File: src/snapster.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      setup?: (element: ElementInterface, edge: Edge) => void
+    }
+  ) {
+    this.grid = new Grid();
+    this.snapping = new Snap({ grid: this.grid, threshold: options.threshold || 8 });
+    this.renderer = new Renderer(
+      {
+        document: options.document,
+        container: options.container,
+        setup: options.setup || ((element, edge) => element.className = `guide guide--${edge.direction}`)
```
Comment (ghost) [inline src/snapster.ts]: Line a bit long.

File: styles/style.css [hunk 2]
```diff
@@ -12,7 +12,7 @@ div {
   width: 800px;
   height: 600px;
   background: #fff;
-  margin: 4rem auto;
+  /* margin: 4rem auto; */
```
Comment (ghost) [inline styles/style.css]: We can uncomment this now.

**Review by ghost — CHANGES_REQUESTED** [review]
Just some formatting tweaks.

**Review by ghost — APPROVED** [review]
🥇

**Review by Willbert — APPROVED** [review]
happy to move on forward :)

**Review by massayoshi — APPROVED** [review]
LGTM

**Pull request merged**